### PR TITLE
slow_conv3d grad_input: Avoid dispatch in parallel region

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66165
* #65759
* #65758
* __->__ #65757
* #65737
* #65726

See gh-56794

Avoid dispatch inside of parallel_for by:
- Replacing Tensor slicing with TensorAccessor
- Replaces `bmm` and `mm` with direct calls to gemm.

Differential Revision: [D31257878](https://our.internmc.facebook.com/intern/diff/D31257878)

### CODE REVIEW

**Comment by ngimel** [comment]
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31257878).

**Comment by ngimel** [comment]
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31257878).

File: aten/src/ATen/native/ConvolutionMM3d.cpp [hunk 1]
```diff
@@ -258,28 +259,32 @@ void slow_conv3d_backward_update_grad_input_frame(
...
+  // Swapping argument order cancels this, since C == AB <=> T(C) == T(B)T(A)
+  const int64_t m = grad_output.size(1) * grad_output.size(2) * grad_output.size(3);
+  const int64_t n = weight.size(1);
+  const int64_t k = weight.size(0) / groups;
+
+  const int64_t lda = m;
+  const int64_t ldb = n;
+  const int64_t ldc = m;
+
+  for (int64_t group = 0; group < groups; ++group) {
```
Comment (ngimel) [inline aten/src/ATen/native/ConvolutionMM3d.cpp:273]: mkl batched gemm (that was used for the grouped case) was an important perf optimization that someone needed. Can we bring it back here? It's not very convenient as it's not part of `cpublas`, but I don't feel comfortable just removing it.

Comment (peterbell10) [reply aten/src/ATen/native/ConvolutionMM3d.cpp:273]: Sure, It might make sense to add a batched gemm to `cpublas`. I'll see how it goes.

**Comment by ngimel** [comment]
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31257878).

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**